### PR TITLE
Add documentation on accessibility.

### PR DESCRIPTION
**Pull request by leerob**
This PR outlines the built-in features of Next.js that help make your application accessible, and will also include some resources for testing/improving accessibility in your app.

Related: https://github.com/vercel/next.js/pull/20428

### CODE REVIEW

File: docs/accessibility.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+The Next.js team is committed to making Next.js accessible to all developers (and their end-users). By adding accessibility features to Next.js by default, we aim to make the Web more inclusive for everyone.
+
+## Route Announcements
+
+When using screen readers or other assistive technology, routes are announced when navigating between pages. This works by default when transitioning between pages on the server (e.g. using the `<a>` tag).
+
+Next.js also supports client-side transitions for improved performance (using `next/link`). To ensure that client-side transitions are also announced to assistive technology, Next.js includes a route announcer by default.
+
+The Next.js route announcer looks for the page name to announce by first inspecting `document.title`, then the `<h1>` element, and finally the URL pathname.
```
Comment (junyper) [inline docs/accessibility.md:15]:
```suggestion
The Next.js route announcer looks for the page name to announce by first inspecting `document.title`, then the `<h1>` element, and finally the URL pathname. For the best accessible user experience, please be sure that each page in your application has a unique and descriptive title.
```

File: docs/accessibility.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+If your application requires JavaScript to be disabled, and only HTML to be used, you can remove all JavaScript from your application using an experimental flag:
+
+```js
+// next.config.js
+export const config = {
+  unstable_runtimeJS: false,
+}
+```
+
+## Accessibility Resources
```
Comment (junyper) [inline docs/accessibility.md:42]: Maybe consider adding [the WebAIM WCAG checklist](https://webaim.org/standards/wcag/checklist), the [WCAG 2.1 Guidelines](https://www.w3.org/TR/WCAG21/), and [The A11y Project](https://www.a11yproject.com/)?

File: docs/upgrading.md [hunk 3]
```diff
@@ -4,6 +4,10 @@ description: Learn how to upgrade Next.js.
 
 # Upgrade Guide
 
+## Upgrading to 12.2
```
Comment (junyper) [inline docs/upgrading.md:7]: This seems unrelated.

**Review by junyper — CHANGES_REQUESTED** [review]
Thank you so much for adding this! I added some suggestions.

File: docs/upgrading.md (hunk 3 above)
Comment (leerob) [reply docs/upgrading.md:7]: It is but I'm sneaking it in here 😂 I noticed this was missing and likely good to have.

File: docs/accessibility.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+description: Learn about the built-in accessibility features of Next.js.
+---
+
+# Accessibility
+
+The Next.js team is committed to making Next.js accessible to all developers (and their end-users). By adding accessibility features to Next.js by default, we aim to make the Web more inclusive for everyone.
+
+## Route Announcements
+
+When using screen readers or other assistive technology, routes are announced when navigating between pages. This works by default when transitioning between pages on the server (e.g. using the `<a>` tag).
```
[the remark this answers is not in the document]
Comment (leerob) [reply docs/accessibility.md:11]: Thank you!

**Review by shuding — APPROVED** [review]
Looks great!
